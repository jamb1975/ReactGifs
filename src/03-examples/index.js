export * from './InvoiceCard'
export * from './LoadingMessage'