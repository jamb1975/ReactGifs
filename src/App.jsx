import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import {heroes, owners} from './data/heroes'

function App2() {
 
  const get_heroe =(id)=>{ 

    return heroes.find(element =>element.id == id);
  }
  
  const [count, setCount] = useState(0)
  const retornoArreglo = () =>{
    return ['ABC', 123];
  }
  const get_heroes_by_own = (own) => heroes.filter(heroe => heroe.owner == own);
  const heroes_by_own = get_heroes_by_own("Marvel");
  console.log(heroes_by_own);
  const   [letras, numeros] = retornoArreglo();
  console.log(letras, numeros);  
  const arregloNomnre = (nombre) =>{
    return [nombre, ()=>{console.log("Hola")}];
  } 
  const [nombre, setNombre] = arregloNomnre('Javi')
  console.log(nombre)
  setNombre();

  //promesas
  const getHeroeById = (id)=>{
    return new Promise((resolve, reject)=>{
      setTimeout(()=>{
        const heroe = get_heroe(id);
        if (heroe) {  return resolve(heroe) }
        else {return reject('No se pudo encontrar el heroe')};
        }, 2000)
    });
}
  getHeroeById(2).then(console.log)
  .catch(console.warn);
  const getImagen = async () =>{
      try{
      const apiKey = 'qTKV8DfFeGMSJje3KT7cieQYBD2oK7SI';
      const httpRequest = await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}`);
      const {data} = await httpRequest.json();
      const {url} = data.images.original;
      //const img = document.getElementById("gif");
      //img.src = url;
      url
      } catch(error){
        console.log(error);
      };
    }

  getImagen();
  const activo = true;
  let msg = ( activo ) ? 'Activo' : 'Inactivo';
  let msg2 = !activo && 'Activo';
  console.log(msg);
  console.log(msg2);
 
 
  return (
    <>
      <div>
        <a href="https://vite.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img id="gif" className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.jsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  )
}

export default App
