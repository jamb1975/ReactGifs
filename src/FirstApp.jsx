//import {Fragment} from "react"
import PropTypes from 'prop-types'
import { useState } from 'react'


export const FirstApp = ( {titulo, subtitulo, name, value} ) => {
   
   // throw new Error("El title no existe")
   const[counter, setCounter] = useState( value )
   const handleAdd = () => setCounter((c)=> c + 1);
   const handleSubtract = () => setCounter((c) => c - 1);
   const reset = () => setCounter(value); 
    return(    
        <> 
            <h1 data-testid="text-title">{ titulo }</h1>
            <p>{ subtitulo }</p>
            <p>{ subtitulo }</p>
            <p>{ name }</p>
            
           
            <code>Counter App</code>
             <p>{counter}</p>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleSubtract }>-1</button>
            <button aria-label='btn-reset' onClick={ reset }>Reset</button>
        </>
    )
}

   FirstApp.PropTypes = {
    value: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    title: "No hay subtitulo",
    subtitle: "No hay subtitulo",
    name: "Javi David"
}