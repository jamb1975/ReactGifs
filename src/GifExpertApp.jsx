import { useState } from "react"
import { AddCategory, GigGrid } from "./components";


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Genesis']);

    const onAddCategory = (newCategory) => {
        console.log(newCategory);
        if (categories.includes(newCategory)) return;
        setCategories( c =>   [ ...c, newCategory, ] );
        
    }
   
    return (
        <>
      
        { /* titulo*/}
            <h1>GifExpertApp</h1>
             
        { /* input */} 
        <AddCategory onNewCategory={ event => onAddCategory(event)}
        />
        { /* Listado de gif*/}
        <ol>
            { categories.map( category => 
                <GigGrid 
                        key= {category} 
                        category={ category } 
                        /> 
            ) }
           
        </ol>

       </>   
    )
}