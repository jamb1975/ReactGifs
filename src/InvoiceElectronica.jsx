/*export function App(){
    return (<h1>Hola Mundo!!!</h1>);
}*/

export const InvoiceElectronica = ()=>(<h1>Hola Mundo  Invoice!!!</h1>)
export const getSaludo = (name) => "Nombre ".concat(name);