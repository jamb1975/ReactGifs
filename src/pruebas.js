import { heroes} from './data/heroes'
export const getSaludo = (name) => "Nombre ".concat(name);
export const getUser = ()=>({
    uid: "ABC123",
    username: "JAMB"

});

export const getUserUnique = (user)=>({
    uid: "ABC123",
    username: user

});

export const retornaArreglo = ()=> ["ABC", 123]
export const get_heroes_by_own = (own) => heroes.filter(heroe => heroe.owner == own);
export const getHeroeById = (id)=> heroes.find( heroe => heroe.id == id);
export const getHeroeByIdAsync = (id)=>{
    return new Promise((resolve, reject)=>{
      setTimeout(()=>{
        const heroe = heroes.find( heroe => heroe.id == id);    
        if (heroe) {  return resolve(heroe) }
        else {return reject('No se pudo encontrar el heroe ' + id)};
        }, 1000)
    });
}
export const getImagen = async (state) =>{
      try{
      const apiKey = 'qTKV8DfFeGMSJje3KT7cieQYBD2oK7SI';
      let httpRequest = "";
      if (state)  httpRequest = await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}`);
      else  httpRequest = await fetch(`https://api.giphy.com/v1/gifs/random?api_key1=${apiKey}`);
      const {data} = await httpRequest.json();
      const {url} = data.images.original;
      //const img = document.getElementById("gif");
      //img.src = url;
      return url
      } catch(error){
        return 'No se encontro la imagen';
      };
    }
