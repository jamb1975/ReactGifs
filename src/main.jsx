import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './GifExpertApp.css';
import {FirstApp} from './FirstApp';
import { GifExpertApp } from './GifExpertApp'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GifExpertApp/>
  </StrictMode>,
)
