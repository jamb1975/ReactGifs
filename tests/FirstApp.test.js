import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {
    test('Debe hacer match con el snapshot', () => {
        const titulo = 'Yo soy Jav';
        const   { container} = render( <FirstApp titulo = { titulo } />);
     expect( container ).toMatchSnapshot();

    });

    test('Debe mostrar el titulo en h1', () => {
        const titulo = 'Yo soy Jav';
     const   { container, getByText, getByTestId } = render( <FirstApp titulo = { titulo } />);
     //expect( getByText( titulo ) ).toBeTruthy();
    /* const h1 = container.querySelector('h1');
     console.log(h1.innerHTML);
     expect(h1.innerHTML).toContain( titulo );*/
     expect(getByTestId('text-title').innerHTML).toBe(titulo);

    });

    test('Debe mostrar el titulo en h1', () => {
     const titulo = 'Yo soy Jav';
     const   { container, getByText, getByTestId } = render( <FirstApp titulo = { titulo } />);
     //expect( getByText( titulo ) ).toBeTruthy();
    /* const h1 = container.querySelector('h1');
     console.log(h1.innerHTML);
     expect(h1.innerHTML).toContain( titulo );*/
     expect(getByTestId('text-title').innerHTML).toBe(titulo);

    });

     test('Debe mostrar el SUBtitulo en h1', () => {
        const titulo = 'Yo soy Jav';
        const subTitulo = 'Learn react is wom';
        const value = 100;
        const   { getAllByText} = render( <FirstApp titulo = { titulo } subtitulo={subTitulo} />);
     //expect( getByText( titulo ) ).toBeTruthy();
    /* const h1 = container.querySelector('h1');
     console.log(h1.innerHTML);
     expect(h1.innerHTML).toContain( titulo );*/
     expect(getAllByText(subTitulo).length).toBe(2);

    });
   
});