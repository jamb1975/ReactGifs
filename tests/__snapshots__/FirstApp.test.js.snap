// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="text-title"
  >
    Yo soy Jav
  </h1>
  <p />
  <p />
  <p />
  <code>
    Counter App
  </code>
  <p />
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
